forgot that offset had to be affected by scale

RobotAlbatross opened the pull request.

RobotAlbatross committed:
forgot that offset had to be affected by scale

OceanTubez reviewed (approved):
w code i like very much

sweeezin commented:
ww

sweeezin reviewed (approved):
w

RobotAlbatross committed:
small fixes

loading screen has slightly better safety. (loading would sometimes be closed but it would still run), nameDisplay now set to what was sent, meaning it is no longer a const variable.
